src/backend/api/Fusion.Resources.Logic/Requests/CanApproveStepHandlerBase.cs

```diff
@@ -0,0 +1,123 @@
+                    var result = await authorizationService.AuthorizeAsync(initiator, request, new ResourceOwnerRequirement(path.Parent(), includeDescendants: true));
+                    isAllowed |= result.Succeeded;
+                }
+
+                if (row.IsResourceOwnerAllowed)
+                {
+                    var result = await authorizationService.AuthorizeAsync(initiator, request, new ResourceOwnerRequirement(request.AssignedDepartment, includeDescendants: false));
+                    isAllowed |= result.Succeeded;
+                }
+            }
+
+            if (row.IsCreatorAllowed)
+            {
+                var result = await authorizationService.AuthorizeAsync(initiator, request, new RequestCreatorRequirement(request.Id));
+                isAllowed |= result.Succeeded;
+            }
+
+            if (row.IsOrgChartTaskOwnerAllowed)
+            {
+                var result = await authorizationService.AuthorizeAsync(initiator, request.Project.OrgProjectId, new TaskOwnerInProjectRequirement());
+                isAllowed |= result.Succeeded;
+            }
+
+            if(request.OrgPositionId.HasValue)
+            {
+                if (row.IsOrgChartWriteAllowed)
+                {
+                    var result = await authorizationService.AuthorizeAsync(initiator, request, OrgPositionAccessRequirement.OrgPositionWrite(request.Project.OrgProjectId, request.OrgPositionId.Value));
+                    isAllowed |= result.Succeeded;
+                }
+                if(row.IsOrgChartReadAllowed)
+                {
+                    var result = await authorizationService.AuthorizeAsync(initiator, request, OrgPositionAccessRequirement.OrgPositionRead(request.Project.OrgProjectId, request.OrgPositionId.Value));
+                    isAllowed |= result.Succeeded;
+                }
+                if (row.IsDirectTaskOwnerAllowed)
+                {
+                    var requirement = new TaskOwnerForPositionRequirement(
+                        request.Project.OrgProjectId, 
+                        request.OrgPositionId.Value,
+                        request.OrgPositionInstance.Id
+                    );
+                    var result = await authorizationService.AuthorizeAsync(initiator, request.Project.OrgProjectId, requirement);
+                    isAllowed |= result.Succeeded;
+                }
+            }
+            
+            if(row.IsOrgAdminAllowed)
+            {
+                var result = await authorizationService.AuthorizeAsync(initiator, request, OrgProjectAccessRequirement.OrgWrite(request.Project.OrgProjectId));
+                isAllowed |= result.Succeeded;
+            }
+
+            isAllowed |= initiator.IsApplicationUser();
+            isAllowed |= initiator.IsInRole("Fusion.Resources.FullControl");
+            isAllowed |= initiator.IsInRole("Fusion.Resources.Internal.FullControl");
+
+            if (!isAllowed) throw new UnauthorizedWorkflowException();
```

Comment:
Should include a message with which steps have been evaluated and the different outcomes? 
I think an informative message to the user could save us from much support requests. 

Should also add more telemetry here.. 
In the api we generate a payload based on the evaluated requirements. Maybe we could to that here as well. We should generate a similar payload and trace to AI, that way we can see whats going on _when_ we get questions.